cpp/src/arrow/engine/substrait/relation_internal.cc

```diff
@@ -94,6 +94,20 @@ Status CheckRelCommon(const RelMessage& rel) {
   return Status::OK();
 }
 
+// Other helper functions
+Status DiscoverFilesFromDir(std::shared_ptr<fs::LocalFileSystem>& local_fs,
+                            std::string dirpath, std::vector<fs::FileInfo>& rel_fpaths) {
```

Comment:
As per the coding conventions, we want to avoid non-const references.
Immutable args should be passed by value or const-ref.
Mutable args should be passed by pointer.
```suggestion
Status DiscoverFilesFromDir(const std::shared_ptr<fs::LocalFileSystem>& local_fs,
                            const std::string& dirpath, std::vector<fs::FileInfo> *rel_fpaths) {
```